lms/models/group_info.py

```diff
@@ -88,3 +90,9 @@ def __repr__(self, _class_name="lms.models.GroupInfo"):
             f" authority_provided_id:'{self.authority_provided_id}'"
             f" consumer_key:'{self.consumer_key}'>"
         )
+
+    @classmethod
+    def metadata_fields(cls):
+        for column in cls.iter_columns():
+            if column.key not in cls.NON_METADATA_FIELDS:
+                yield column.key
```

Comment:
I'm having trouble figure out what a "metadata" field is vs a "non-metadata" field, semantically. Might need another name? Or a docstring?

Reply:
I think the difference is that the non-metadata fields are the ones that identify the row (id, authority_provided_id, and consumer_key) whereas the metadata fields are the ones that _don't_ identify the row and that might actually be changed on an update, and come from `request.params`. I can't think of a better name for it but maybe just an explanatory docstring would help

Reply:
This might be named backwards, they are really the data fields. The `metadata` is presumably `id` etc.